src/bun.js/bindings/JSSocketAddressDTO.cpp

```diff
@@ -24,40 +28,47 @@ Structure* createStructure(VM& vm, JSGlobalObject* globalObject)
         JSC::Identifier::fromString(vm, "address"_s),
         0,
         offset);
+    ASSERT(offset == addressOffset);
 
     structure = structure->addPropertyTransition(
         vm,
         structure,
         JSC::Identifier::fromString(vm, "family"_s),
         0,
         offset);
+    ASSERT(offset == familyOffset);
 
     structure = structure->addPropertyTransition(
         vm,
         structure,
         JSC::Identifier::fromString(vm, "port"_s),
         0,
         offset);
+    ASSERT(offset == portOffset);
 
     return structure;
 }
 
 } // namespace JSSocketAddress
 } // namespace Bun
 
-extern "C" JSObject* JSSocketAddress__create(JSGlobalObject* globalObject, JSString* value, int32_t port, bool isIPv6)
+extern "C" JSC__JSValue JSSocketAddressDTO__create(JSGlobalObject* globalObject, JSString* address, int32_t port, bool isIPv6)
 {
-    static const NeverDestroyed<String> IPv4 = MAKE_STATIC_STRING_IMPL("IPv4");
-    static const NeverDestroyed<String> IPv6 = MAKE_STATIC_STRING_IMPL("IPv6");
 
     VM& vm = globalObject->vm();
-
     auto* global = jsCast<Zig::GlobalObject*>(globalObject);
 
-    JSObject* thisObject = constructEmptyObject(vm, global->JSSocketAddressStructure());
-    thisObject->putDirectOffset(vm, 0, value);
-    thisObject->putDirectOffset(vm, 1, isIPv6 ? jsString(vm, IPv6) : jsString(vm, IPv4));
-    thisObject->putDirectOffset(vm, 2, jsNumber(port));
+    ASSERT(port < std::numeric_limits<uint16_t>::max());
 
-    return thisObject;
+    JSObject* thisObject = constructEmptyObject(vm, global->JSSocketAddressDTOStructure());
+    thisObject->putDirectOffset(vm, Bun::JSSocketAddressDTO::addressOffset, address);
+    thisObject->putDirectOffset(vm, Bun::JSSocketAddressDTO::familyOffset, isIPv6 ? jsString(vm, String(IPv6)) : jsString(vm, String(IPv4)));
+    thisObject->putDirectOffset(vm, Bun::JSSocketAddressDTO::portOffset, jsNumber(port));
+
+    return JSValue::encode(thisObject);
 }
+
+WTF::StaticStringImpl* const IPv4 = MAKE_STATIC_STRING_IMPL("IPv4");
```

Comment:
We should not be creating these initializers at application start. This runs when you do `bun --version`.

Instead, we should use `BunCommonStrings.h` and put it in there.